### PR TITLE
SAA-678: Include case note text in attendance history

### PR DESCRIPTION
**Pull request by gerryallan-moj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/util/TransformFunctions.kt [hunk 1]
```diff
@@ -341,6 +342,7 @@ fun transform(attendanceHistory: EntityAttendanceHistory): ModelAttendanceHistor
         notes = it.notes,
       )
     },
+    caseNoteText = attendanceHistory.caseNoteId?.let { caseNotesApiClient.getCaseNote(prisonerNumber, attendanceHistory.caseNoteId)?.text },
     comment = attendanceHistory.comment,
```
Comment (timharrison-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/util/TransformFunctions.kt:346]: I'm not sure about this change, Gerry..  feels a bit off to pass an API client down into the transform functions to make the call to get case notes from there?  For me, these calls feel better being made in a service method, and passed into the transform functions as pieces of data, but I'll ask Matt for a second opinion and get back to you.

Comment (matt-crowson-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/util/TransformFunctions.kt:346]: Yes agreed.  Should really only be using the api clients in services/component stereotypes.

Comment (timharrison-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/util/TransformFunctions.kt:346]: Yeah, Matt agrees with me - sorry!  Is there any way the calls to the case note service could stay within the service methods, and passed as data into the transforms instead?

Comment (gerryallan-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/util/TransformFunctions.kt]: Reworked as suggested

**Review by matt-crowson-moj — APPROVED** [review]
LGTM

**Pull request merged**